server/game/cards/09.2-BoB/IsawaTadaka2.js

```diff
@@ -12,13 +12,18 @@ class IsawaTadaka2 extends DrawCard {
                 location: Locations.DynastyDiscardPile,
                 mode: TargetModes.Unlimited
             }),
-            gameAction: AbilityDsl.actions.cardMenu(context => ({
-                cards: context.player.opponent && context.costs.removeFromGame ?
-                    context.player.opponent.hand.shuffle().slice(0, context.costs.removeFromGame.length) :
-                    [context.source],
-                gameAction: AbilityDsl.actions.discardCard(),
-                message: '{0} chooses {1} to be discarded from {2}',
-                messageArgs: (card, player, cards) => [player, card, cards]
+            gameAction: AbilityDsl.actions.ifAble(() => ({
```

Comment:
I'm not sure card implementations should be expected to be able to deal with this explicitly.  I think the game engine should handle this case better

Reply:
so the issue with Tadaka is that we do the "look at" and "discard" in the same step; if the discard is illegal, the whole step is skipped and you don't get to look at any cards.  Is there a way to do this like Investigator where there's a "reveal" action first, which randomly selects the cards, and then does the discard action using the same cards as in the reveal action?  I split it up because I didn't know if you could pass the subarray to the next action.